src/NuGet.Core/NuGet.Versioning/VersionRangeFactory.cs

```diff
@@ -13,6 +13,10 @@ public partial class VersionRange
     {
         // Static factory methods for creating version range objects.
 
+        // Cached mappings from (string value, bool allowFloating) => VersionRange. On cache hit, avoids allocations during TryParse.
+        private static Dictionary<(string, bool), VersionRange> ParsedVersionRangeMapping = new Dictionary<(string, bool), VersionRange>();
+        private const int ParsedVersionRangeMappingMaxEntries = 500;
```

Comment:
Same comments about the max entries const and dictionary initial capacity as I had in the other file.

Reply:
Sorry about that, I should have fixed that as part of the last commit.